src/py42/services/_auth.py

```diff
@@ -41,3 +41,12 @@ def _get_credentials(self):
         headers = {"totp-auth": str(current_token)} if current_token else None
         response = self._auth_connection.get(uri, params=params, headers=headers)
         return u"v3_user_token {}".format(response["v3_user_token"])
+
+
+class JWTAuth(C42RenewableAuth):
```

Comment:
Technically, we were using JWT auth prior as well (`V3Auth`), I am not sure about the name of this class.

Reply:
Yea, the ticket description used words like `jwt_provider` etc due to which I came with the name. What about `CustomAuth`

Reply:
What about `JWTDirectAuth`? Since all this class is doing is taking a token directly and using it. It pushes all the "get a token" logic outside to the provider func.

Reply:
`JWTDirect` or `DirectJWT` sound good here.

Reply:
Or perhaps `CustomJWTAuth`? This class takes a method that gets a token, not a token directly. I think `Custom` captures the "user specified" part, while `JWT` indicates the restriction that that method must return a JWT.

Reply:
changed.